### ISSUE TITLE
New release

### ISSUE THREAD

**Issue by ldv-alt** [linux-pam/linux-pam#690]
Given that we've had quite a few various fixes and cleanups applied since the previous release, I'd like to make a new release so that more users could benefit from it.  Let's do it on some Wednesday of January, say, on 10th or 17th.

@stoeckmann @thkukuk @t8m, What do you think about it?

**Comment by BenBE** [comment]
Given that there's still a bit of PRs coming in re cleanup I'd suggest going for the later of the two. This also allows for admins to get back into work, before they need to cope with a PAM update for all their systems.

Furthermore as many of the recent changes bear high risk for regressions this gives further time for people to test the changes and look for breakage.

**Comment by stoeckmann** [comment]
I'd go with "release early, release often."

Although I have some more things I'd like to add, mostly getting rid of fgets completely, having way too many changes between releases might make it more difficult to track down which change exactly triggered a regression. Hopefully there will be none, of course.

So if it's about being "feature complete" in the sense that I'm done with everything: Not yet. We can see how far we can progress until the mentioned dates.

But as stated in the beginning, I would be fine with a new release at any given day.

**Comment by ldv-alt** [comment]
@ikerexxe @thesamesam, any preferences downstream?

**Comment by ikerexxe** [comment]
As long as it's before February 13th, I'm fine with anything. That's one week before the Fedora Beta freeze and it should give me ample time to rebase and build.

> So if it's about being "feature complete" in the sense that I'm done with everything: Not yet. We can see how far we can progress until the mentioned dates.

Having all those changes merged would be nice. I see they do some refactoring of the existing code, and having all of them at once will help with the future maintenance of the downstream code.

@stoeckmann do you have any idea of how many changes are missing for the "feature complete"? Or are they all already available for review?

**Comment by stoeckmann** [comment]
> @stoeckmann do you have any idea of how many changes are missing for the "feature complete"? Or are they all already available for review?

The `pam_env` adjustments are missing yet. I would like to do them in two PRs, unfortunately the second one will depend on the first. The first one should be available in the next few hours, the second one is pretty much a copy&paste of pam_assemble_line out of libpam into pam_env, with a minor adjustment. This would take almost no time.

From a pure development perspective, it'll be done today.

**Comment by stoeckmann** [comment]
See https://github.com/linux-pam/linux-pam/pull/696 and https://github.com/linux-pam/linux-pam/pull/697

**Comment by thkukuk** [comment]
> Given that we've had quite a few various fixes and cleanups applied since the previous release, I'd like to make a new release so that more users could benefit from it. Let's do it on some Wednesday of January, say, on 10th or 17th.

Sorry, I was on FTO travelling the last two weeks.
I'm in favor of a "quick" release, I'm pretty sure with all the changes we have to make another release shortly anyways to fix the regressions or new bugs.

**Comment by stoeckmann** [comment]
> I'm pretty sure with all the changes we have to make another release shortly anyways to fix the regressions or new bugs.

Are enough features involved that it could be a 1.6.0 version? Less people would expect flawless software in a .0 release. Not sure if anyone really cares about -rc releases, otherwise that would be an option, too.

**Comment by ldv-alt** [comment]
> Are enough features involved that it could be a 1.6.0 version? Less people would expect flawless software in a .0 release.

I'd say support of arbitrary long lines is big enough for .0 release.

> Not sure if anyone really cares about -rc releases, otherwise that would be an option, too.

Unfortunately, -rc releases do not attract enough attention nowadays.

**Comment by ldv-alt** [comment]
> > Given that we've had quite a few various fixes and cleanups applied since the previous release, I'd like to make a new release so that more users could benefit from it. Let's do it on some Wednesday of January, say, on 10th or 17th.
> 
> Sorry, I was on FTO travelling the last two weeks. I'm in favor of a "quick" release, I'm pretty sure with all the changes we have to make another release shortly anyways to fix the regressions or new bugs.

OK, let's stick with 17th then.

**Comment by stoeckmann** [comment]
> I'd say support of arbitrary long lines is big enough for .0 release.

Yes, we can argue that configuration files which work with 1.6.0 do not necessarily lead to the same results with 1.5.3. Just as reported in https://github.com/linux-pam/linux-pam/issues/532.

**Comment by ldv-alt** [comment]
@stoeckmann, could you summarize user-visible changes you made for this release that are worth mentioning in the NEWS file, please?

**Comment by stoeckmann** [comment]
How about this:

libpam: Use getrandom(2) as source of randomness if available.
libpam: Fix calculation of fail delay with very long delays.
pam_access: Fix truncation of very long group names.
pam_echo: Fix file handling to prevent overflows and short reads.
pam_env: Support '\' character in environment variable values.
pam_exec: Fix stack overflow with binary output of programs.
pam_faildelay: Validate parameter ranges.
pam_listfile: Treat \r and \n exactly the same in configuration.
pam_mkhomedir: Harden directory creation against timing attacks. Please not that using *at functions leads to more open file handles during creation.
pam_nologin: Fix file handling to prevent short reads.
pam_shells: Only allow shell entries with absolute paths.
pam_succeed_if: Do not treat empty strings as numerical value 0.
pam_unix: Support disabled password aging.
pam_unix: Synchronize password aging with shadow.
pam_unix: Validate string to number conversions.
pam_unix: Fix truncation of very long user names.

Support configuration files with arbitrarily long lines.
Allow build of Linux-PAM outside of its source tree.

**Comment by BenBE** [comment]
Furthermore:

libpam: avoid infinite recursion with includes
pam_xauth: avoid building argv inside run_coprocess

**Comment by ldv-alt** [comment]
> pam_xauth: avoid building argv inside run_coprocess

I don't think run_coprocess is user visible.

**Comment by stoeckmann** [comment]
I've updated the list of changes to also reflect latest `pam_pwhistory` and `pam_unix` changes regarding opasswd.

**Comment by ldv-alt** [comment]
Thanks!
